protocol/transport/cache.go

```diff
@@ -0,0 +1,81 @@
+package transport
```

Comment:
I'm wondering if `cache` is a best naming for this functionality. I would suspect from name that it contains some messages for quick access. But in fact we keep track of processed messages here. wdyt?

Reply:
If we don't mind being verbose, we could call it:
`ProcessedMessageIDsCache` 
what do you think?

Reply:
super) I love verbose names)